squad/ci/backend/lava.py

```diff
@@ -327,8 +328,10 @@ def __parse_results__(self, data, test_job):
                         if metadata['error_type'] in ['Infrastructure', 'Lava', 'Job']:
                             completed = False
                         # automatically resubmit in some cases
-                        if metadata['error_type'] in ['Infrastructure', 'Job'] and \
-                                any(substring.lower() in metadata['error_msg'].lower() for substring in lava_infra_error_messages):
+                        infra_messages_re = re.compile("|".join(lava_infra_error_messages))
+                        if metadata['error_type'] in ['Infrastructure', 'Job', 'Test'] and \
+                                len(lava_infra_error_messages) > 0 and \
+                                infra_messages_re.search(metadata['error_msg']) is not None:
```

Comment:
are you dropping the case insensitivity (`.lower()`) on purpose?

Reply:
you should probably handle invalid regular expressions

Reply:
I'm in doubt wrt joining all the items like with (`"|".join(...)`). if one of the items is not a valid regular expression, you will invalidate the entire check. I would keep testing one by one, and ignoring the ones that are not valid regexes.